DESCRIPTION

```diff
@@ -37,6 +37,7 @@ Imports:
     xml2 (>= 1.0.0),
     xmlparsedata (>= 1.0.5)
 Suggests:
+    bookdown,
```

Comment:
is there any way for us to avoid these Suggests?

e.g. could we detect namespaces in the file linted, and attempt `requireNamespace()` on them, thereby obviating a Suggested dependency, as long as the calling user has the package available?

Doesn't have to be done in this PR (e.g. if it requires some substantial machinery changing in `get_source_expressions()` or `lint()`), just wondering if it's possible.

Reply:
This might work, but not always. 

Because, more often than not, people won't use namespace while specifying output formats. That is, they are more likely to write 

```yaml
---
output: tufte_html
---
```

instead of 

```yaml
---
output: tufte::tufte_html
---
```

This is a result of habits formed while using the more standard output formats where users write 

```yaml
---
output: html_document
---
```

instead of 

```yaml
---
output: rmarkdown::html_document
---
```

Reply:
I'm that case, how does knitr detect the namespace?

Reply:
could you file a follow-up issue for someone to look into this eventually? thanks

Reply:
> [In] that case, how does knitr detect the namespace?

Yeah, I was also curious about the same and went down the rabbit hole of `rmarkdown::render()` and `knitr::knit()`, but gave up given the complexity of these functions. Might revisit later when I have a bit more time on my hands.

> could you file a follow-up issue for someone to look into this eventually? thanks

Done: https://github.com/r-lib/lintr/issues/1617